### DISCUSSION TITLE
Spore Meeting Agenda @ 2024-01-02

### DISCUSSION THREAD

**Discussion by ashuralyk** [sporeprotocol/spore-scrum#11] [Show and tell]
## Goal
1. integrate co-build into `spore-contract` and `spore-sdk` as fast as possible, see discussion here: [papp-architecture](https://discord.com/channels/674518700518408213/1169476459044286494)
a. for the integration with contract: https://github.com/cryptape/ckb-transaction-cobuild-poc/blob/main/contracts/transaction-cobuild-lock-demo/src/entry.rs#L20
b. for the integration with sdk: https://github.com/cryptape/ckb-transaction-cobuild-poc/tree/main/dapp

2. prepare `spore-sdk` standalone CI test as fast as possible.

3. talks:
a. how do we launch mainnet from Spore contract v2?
b. do we need to launch mainnet just from Spore contract v1?
c. what if we lost our previous contract v1 commit? https://github.com/sporeprotocol/spore-contract/issues/24

## Previous Work
* solve bugs in Spore contract v2 
  - [x] https://github.com/sporeprotocol/spore-contract/pull/20
  * `MemWriteOnExecutablePage` error
    - [ ] https://github.com/sporeprotocol/spore-contract/issues/25
    - [ ] solution ideas from `by`:   [需要把 CKBDLContext 放在堆上](https://discord.com/channels/674518700518408213/852365135791128648/1109765873356525588)
 
* integrate Spore features v2 into Spore SDK and solve bugs
  - [ ] https://github.com/sporeprotocol/spore-sdk/pull/51

* prepare standalone native CKB node env for the CI test of Spore SDK
  - [ ] https://github.com/sporeprotocol/spore-sdk/issues/52
  - [ ] https://github.com/sporeprotocol/spore-sdk/pull/48

* prepare e2e test for Spore Demo
  - [x] write test cases brain map https://mubu.com/doc/46ygy0TIl3M
  - [ ] https://github.com/sporeprotocol/spore-demo/issues/34

## Next Plan
* complete standalone native CKB node env for Spore SDK, manual test is insufficient
* lean about how to integrate co-build for both contract and sdk
* if launch mainnet from Spore contract v2
  * start a new spore demo to support latest deployed Spore contract v2
  * only enable features of Spore and Cluster, despite Agent, Proxy and Mutant
* receive 100k CKB from Jan

---------

please write down your last works in the comment following this template:
```
Done:
1. ..
2. ..

Next:
1. ..
2. ..
```

cc @RongxinZhang @ShookLyngs @Dawn-githup

**Comment by ShookLyngs** [comment]
Done:
1. V2 related updates: https://github.com/sporeprotocol/spore-sdk/pull/55
2. A minor fix of ClusterProxy: https://github.com/sporeprotocol/spore-sdk/pull/51

Next:
1. Co-build study and intergration: https://github.com/sporeprotocol/spore-sdk/issues/53
3. Rewrite major features' tests for automation: https://github.com/sporeprotocol/spore-sdk/issues/54

**Comment by Dawn-githup** [comment]
Done:
  
  1. Simplify the spore-dev process: https://github.com/Dawn-githup/spore_devenv
  2. spore-sdk #51 test：[fix(sdk): fix ClusterProxyArgs.minPayment type spore-sdk#51](https://github.com/sporeprotocol/spore-sdk/pull/51)
  3. spore-sdk #55 review: https://github.com/sporeprotocol/spore-sdk/pull/55 
  
Next:

  1. spore-demo e2e functional test: https://mubu.com/doc/46ygy0TIl3M
  2. spore-dev optimizes details and adds other features：https://github.com/cryptape/acceptance-internal/issues/619

**Comment by ashuralyk** [comment]
# Progress of scrum
1. co-build integration in `spore-contract` has completed
2. the standalone Github CI env of `spore-sdk` has prepared, leaves some bugs to solve
